### PR TITLE
fix broken studio links

### PR DESCRIPTION
**Pull request by queerviolet**
this PR aims to collect correct definitions for all schemas under the `https://specs.apollo.dev` namespace along with minimal acceptable HTML documentation (definitions are correct, no broken links)

schemas defined:

- [core/v0.1](/apollographql/specs/blob/fix-broken-studio-links/core/v0.1/core-v0.1.graphql)
- [core/v0.2](/apollographql/specs/blob/fix-broken-studio-links/core/v0.2/core-v0.2.graphql)
- [federation/v1.0](/apollographql/specs/blob/fix-broken-studio-links/federation/v1.0/federation-v1.0.graphql)
- [federation/v2.0](/apollographql/specs/blob/fix-broken-studio-links/federation/v2.0/federation-v2.0.graphql)
- [inaccessible/v0.1](/apollographql/specs/blob/fix-broken-studio-links/inaccessible/v0.1/inaccessible-v0.1.graphql)
- [inaccessible/v0.2](/apollographql/specs/blob/fix-broken-studio-links/inaccessible/v0.2/inaccessible-v0.2.graphql)
- [join/v0.1](/apollographql/specs/blob/fix-broken-studio-links/join/v0.1/join-v0.1.graphql)
- [join/v0.2](/apollographql/specs/blob/fix-broken-studio-links/join/v0.2/join-v0.2.graphql)
- [link/v1.0](/apollographql/specs/blob/fix-broken-studio-links/link/v1.0/link-v1.0.graphql)
- [tag/v0.1](/apollographql/specs/blob/fix-broken-studio-links/tag/v0.1/tag-v0.1.graphql)
- [tag/v0.2](/apollographql/specs/blob/fix-broken-studio-links/tag/v0.2/tag-v0.2.graphql)

documentation previews:

- [core/v0.1](https://deploy-preview-21--apollo-specs.netlify.app/core/v0.1)
- [core/v0.2](https://deploy-preview-21--apollo-specs.netlify.app/core/v0.2)
- [federation/v1.0](https://deploy-preview-21--apollo-specs.netlify.app/federation/v1.0)
- [federation/v2.0](https://deploy-preview-21--apollo-specs.netlify.app/federation/v2.0)
- [inaccessible/v0.1](https://deploy-preview-21--apollo-specs.netlify.app/inaccessible/v0.1)
- [inaccessible/v0.2](https://deploy-preview-21--apollo-specs.netlify.app/inaccessible/v0.2)
- [join/v0.1](https://deploy-preview-21--apollo-specs.netlify.app/join/v0.1)
- [join/v0.2](https://deploy-preview-21--apollo-specs.netlify.app/join/v0.2)
- [link/v1.0](https://deploy-preview-21--apollo-specs.netlify.app/link/v1.0)
- [tag/v0.1](https://deploy-preview-21--apollo-specs.netlify.app/tag/v0.1)
- [tag/v0.2](https://deploy-preview-21--apollo-specs.netlify.app/tag/v0.2)

### CODE REVIEW

**Comment by queerviolet** [comment]
hi if i tagged you as a reviewer on this PR it's because i believe you maintain software which cares about one or more of these schemas:

- [core/v0.1](https://github.com/apollographql/specs/blob/fix-broken-studio-links/core/v0.1/core-v0.1.graphql)
- [core/v0.2](https://github.com/apollographql/specs/blob/fix-broken-studio-links/core/v0.2/core-v0.2.graphql)
- [federation/v1.0](https://github.com/apollographql/specs/blob/fix-broken-studio-links/federation/v1.0/federation-v1.0.graphql)
- [federation/v2.0](https://github.com/apollographql/specs/blob/fix-broken-studio-links/federation/v2.0/federation-v2.0.graphql)
- [inaccessible/v0.1](https://github.com/apollographql/specs/blob/fix-broken-studio-links/inaccessible/v0.1/inaccessible-v0.1.graphql)
- [inaccessible/v0.2](https://github.com/apollographql/specs/blob/fix-broken-studio-links/inaccessible/v0.2/inaccessible-v0.2.graphql)
- [join/v0.1](https://github.com/apollographql/specs/blob/fix-broken-studio-links/join/v0.1/join-v0.1.graphql)
- [join/v0.2](https://github.com/apollographql/specs/blob/fix-broken-studio-links/join/v0.2/join-v0.2.graphql)
- [link/v1.0](https://github.com/apollographql/specs/blob/fix-broken-studio-links/link/v1.0/link-v1.0.graphql)
- [tag/v0.1](https://github.com/apollographql/specs/blob/fix-broken-studio-links/tag/v0.1/tag-v0.1.graphql)
- [tag/v0.2](https://github.com/apollographql/specs/blob/fix-broken-studio-links/tag/v0.2/tag-v0.2.graphql)

it would be a great help if you could verify that the definitions are in line with what your software expects. (the definitions are largely extracted from our software, so that should be the case, but… let's just check)

@sachindshinde i'm less sure about [inaccessible/v0.2](https://github.com/apollographql/specs/blob/fix-broken-studio-links/inaccessible/v0.2/inaccessible-v0.2.graphql), so would particularly appreciate your eyes on that.

File: inaccessible/v0.2/inaccessible-v0.2.graphql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+directive @inaccessible on
+  | FIELD_DEFINITION
+  | OBJECT
+  | INTERFACE
+  | UNION
+  | ENUM
+  | ENUM_VALUE
```
Comment (sachindshinde) [inline inaccessible/v0.2/inaccessible-v0.2.graphql:7]: Slight update to this definition
```suggestion
  | ENUM_VALUE
  | SCALAR
  | INPUT_OBJECT
  | INPUT_FIELD_DEFINITION
  | ARGUMENT_DEFINITION
```

File: federation/v2.0/federation-v2.0.graphql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+directive @key(fields: FieldSet!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE
+directive @requires(fields: FieldSet!) on FIELD_DEFINITION
+directive @provides(fields: FieldSet!) on FIELD_DEFINITION
+directive @external on OBJECT | FIELD_DEFINITION
+directive @shareable on FIELD_DEFINITION | OBJECT
+directive @extends on OBJECT | INTERFACE
+directive @override(from: String!) on FIELD_DEFINITION
+directive @inaccessible on
```
Comment (pcmanus) [inline federation/v2.0/federation-v2.0.graphql:8]: In the implem, `federation/v2.0` essentially gives you `inaccessible/v0.2` and `tag/v0.2`. It seems to be missing the `@tag` definition and the `@inaccessible` locations should match that of v0.2.

File: federation/v2.0/federation-v2.0.graphql [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  | FIELD_DEFINITION
+  | OBJECT
+  | INTERFACE
+  | UNION
+scalar _FieldSet
```
Comment (pcmanus) [inline federation/v2.0/federation-v2.0.graphql:13]: We define the type as `FieldSet` for fed2, without the `_` (there is special code when auto-migrating fed1 schema to fed2 to handle this properly if necessary).

File: join/v0.2/join-v0.2.graphql [hunk 4]
```diff
@@ -0,0 +1,25 @@
+enum Graph
+
+directive @type(
+  graph: Graph!,
+  key: String!,
+  extension: Boolean!
+  resolvable: Boolean!
+) repeatable on OBJECT | INTERFACE
```
Comment (pcmanus) [inline join/v0.2/join-v0.2.graphql:8]: `@type` is allowed on all types (so `OBJECT | INTERFACE | UNION | ENUM | INPUT_OBJECT | SCALAR`).

File: join/v0.2/join-v0.2.graphql [hunk 5]
```diff
@@ -0,0 +1,25 @@
+enum Graph
+
+directive @type(
+  graph: Graph!,
+  key: String!,
```
Comment (pcmanus) [inline join/v0.2/join-v0.2.graphql:5]: Technically the type of `key` here and of `requires` and `provides` in `@field` is a `FieldSet` scalar that is part of the spec too.

But also, `key` is nullable in join v0.2.

File: join/v0.2/join-v0.2.graphql [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  extension: Boolean!
```
Comment (pcmanus) [inline join/v0.2/join-v0.2.graphql:6]: Fwiw, `extension` has a default value of `false` and `resolvable` a default value of `true`.

File: join/v0.2/join-v0.2.graphql [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+directive @field(
+  graph: Graph,
+  requires: String,
+  provides: String,
+  type: String,
+  external: Boolean,
+  override: String,
+  usedOverridden: Boolean      
+) on FIELD_DEFINITION
```
Comment (pcmanus) [inline join/v0.2/join-v0.2.graphql:18]: v0.2 allows `@field` on `INPUT_FIELD_DEFINITION`.

File: federation/v2.0/federation-v2.0.graphql (hunk 2 above)
Comment (queerviolet) [reply federation/v2.0/federation-v2.0.graphql:8]: done

File: federation/v2.0/federation-v2.0.graphql (hunk 3 above)
Comment (queerviolet) [reply federation/v2.0/federation-v2.0.graphql:13]: done

(we should update it [here too](https://www.apollographql.com/docs/federation/federation-spec/#scalar-_fieldset))

File: join/v0.2/join-v0.2.graphql (hunk 4 above)
Comment (queerviolet) [reply join/v0.2/join-v0.2.graphql:8]: done

File: join/v0.2/join-v0.2.graphql (hunk 6 above)
Comment (queerviolet) [reply join/v0.2/join-v0.2.graphql:9]: done

File: join/v0.2/join-v0.2.graphql (hunk 7 above)
Comment (queerviolet) [reply join/v0.2/join-v0.2.graphql:21]: done

File: federation/v1.0/federation-v1.0.graphql [hunk 8]
```diff
@@ -0,0 +1,5 @@
+directive @key(fields: FieldSet!) repeatable on OBJECT | INTERFACE
+directive @requires(fields: FieldSet!) on FIELD_DEFINITION
+directive @provides(fields: FieldSet!) on FIELD_DEFINITION
+directive @external on OBJECT | FIELD_DEFINITION
+scalar _FieldSet
```
Comment (benweatherman) [inline federation/v1.0/federation-v1.0.graphql:5]: **observation** Should the `FieldSet`s in here be `_FieldSet`? Or should this be `_FieldSet`? I think it's the former...

**Review by benweatherman — APPROVED** [review]
Maybe 1 more small change, but otherwise this looks good to me. Great we'll finally have non-broken links!

File: federation/v1.0/federation-v1.0.graphql (hunk 8 above)
Comment (queerviolet) [reply federation/v1.0/federation-v1.0.graphql:5]: oop, fixed

**Pull request merged**